### PR TITLE
Fixes typos in various documentation files

### PR DESCRIPTION
**Pull request by yongjianchn**

### CODE REVIEW

**Comment by randall** [comment]
[approve ci]

File: proxy/hdrs/HdrHeap.h [hunk 1]
```diff
@@ -203,7 +203,7 @@ class HdrHeap
   void set_ronly_str_heap_end(int slot, const char *end);
 
   // Lock read only str heaps so that can't be moved around
-  //  by a heap consolidation.  Does NOT lock for Multi-Threaed
+  //  by a heap consolidation.  Does NOT lock for Multi-Thread
```
Comment (mlibbey) [inline proxy/hdrs/HdrHeap.h]: I think this may be Thread*ed*.

**Comment by scw00** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by randall** [comment]
@yongjianchn Apologies, I was trying to help out here.

**Comment by yongjianchn** [comment]
@randall Thanks for your helping and sorry for my late reply. What else do I need to do about this pr? Should I make a rebase and a force push?

**Comment by zwoop** [comment]
Yes rebase from master and force push

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
[approve ci docs]

**Comment by yongjianchn** [comment]
CI failed to build docs. But I think the problem is not in this commit but in the CI environment. 
I don't know who is able to fix it. 
The error message is:
```
checking for sphinx version >= 1.7.5... No module named 'sphinx'
configure: error: Doc building disabled, check sphinx installation
```

**Comment by randall** [comment]
[approve ci docs]

**Comment by randall** [comment]
[approve ci docs]

**Comment by zwoop** [comment]
Moving out from 8.1.x, unless someone cares very strongly here and makes a PR for the 8.1.x branch (this doesn't cherry-pick cleanly).

**Pull request merged**